bookmarks/api/serializers.py

```diff
@@ -42,7 +44,11 @@ def create(self, validated_data):
         bookmark.title = validated_data['title']
         bookmark.description = validated_data['description']
         tag_string = build_tag_string(validated_data['tag_names'])
-        return create_bookmark(bookmark, tag_string, self.context['user'])
+        bookmark = create_bookmark(bookmark, tag_string, self.context['user'])
+
+        if validated_data['is_archived']:
+            bookmark = archive_bookmark(bookmark)
+        return bookmark
```

Comment:
```suggestion
        bookmark.is_archived = validated_data['is_archived']
        return create_bookmark(bookmark, tag_string, self.context['user'])
```
No need to reuse `archive_bookmark` at this point. Can also remove the import afterwards.